testsrc/org/mozilla/javascript/tests/MemberBoxCallTest.java

```diff
@@ -0,0 +1,108 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.javascript.tests;
+
+import org.junit.Test;
+import org.junit.Before;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import org.mozilla.javascript.*;
+import org.mozilla.javascript.annotations.*;
+
+public class MemberBoxCallTest {
+
+    Scriptable scope;
+
+    @Test
+    public void testPrototypeProperty() {
+		Context cx = Context.enter();
+		try {
+			assertEquals(evaluate(cx, 
```

Comment:
I see a few of properties and functions on "AnnotatedHostObject," but it looks like this test is only checking for "foo." Is there a reason why we can't test the other properties? Similarly, are you sure that all the code paths in ScriptableObject are being exercised here for both Member and MemberBox?